pandas/core/groupby/generic.py

```diff
@@ -1760,6 +1763,19 @@ def value_counts(
             )
             result_series = cast(Series, gb.size())
 
+            # GH-46357 Include non-observed categories
+            # of non-grouping columns regardless of `observed`
+            if any(
+                isinstance(grouping.grouping_vector, (Categorical, CategoricalIndex))
+                and not grouping._observed
+                for grouping in groupings
+            ):
+                levels_list = [ping.group_index for ping in groupings]
+                index, _ = MultiIndex.from_product(
+                    levels_list, names=[ping.name for ping in groupings]
+                ).sortlevel()
+                result_series = result_series.reindex(index, fill_value=0)
+
```

Comment:
I also tried to make this change directly in `_reindex_output` method, so it handles the fact that groupings might have a different observed attribute, but it breaks 60 tests.

Reply:
Makes sense that reindex_output wouldn't work here; the schema for groupby.value_counts differs from other ops.

Reply:
Does L1761 `observed=self.observed` actually do anything? I'd hazard a guess that the groupby code is just using the `observed=False` from L1752. If that is the case, can you make L1761 set observed to False as well for readability.

Reply:
`observed=self.observed` is needed for categorical grouping columns. `observed=False` is used L1752 so every category of any categorical non-grouping column is included in the result.

Reply:
Ah, thanks - I forgot the above groupers on L1752 are only the non-grouping columns.